addons/mrp/static/src/js/mrp_bom_report.js

```diff
@@ -242,6 +257,37 @@ var MrpBomReport = stock_report_generic.extend({
             $parent.remove();
         });
     },
+    _unfoldAll: function () {
+        var self = this;
+        let promises = [];
+        self.$('.o_mrp_bom_report_line').find('.o_mrp_bom_unfoldable').each(function(index, folded) { 
+            promises.push(self[$(folded).data('function')]({currentTarget: folded}));
+        });
+        Promise.all(promises).then(values => {
+            if (values.length) {
+                self._unfoldAll();
+            }
+        });
+    },
+    _unfoldIds: function(ids) {
+        var self = this;
+        let promises = [];
+        self.$('.o_mrp_bom_report_line').find('.o_mrp_bom_unfoldable').each(function(index, folded) { 
+            let id = $(folded).closest('tr').data('id');
+            if (ids.includes(id)) {
+                promises.push(self[$(folded).data('function')]({currentTarget: folded}));
+                ids.splice(ids.indexOf(id), 1);
+            }
+        });
+        Promise.all(promises).then(values => {
+            if (ids.length) {
+                self._unfoldIds(ids);
+            }
+            else {
+                self.$('.o_content').css("display", "inherit");
+            }
+        });
+    },
```

Comment:
That said, I think it should be better to work with bom line's id (instead of product's id) or with a tag to know if a line is folded or not.
Because while working with the product's id, the folding/unfolding methods won't work properly if the same product (with a BoM) is in multiple places in the report.
But it seems it's already the case in v15, so maybe a fix is more appropriate regarding this issue (I'll check that with @mathieu-mgm ).